baremetal/src/Runtime.cc

```diff
@@ -55,16 +59,48 @@ class Connection : public ebbrt::TcpHandler {
 }  // namespace
 
 void ebbrt::runtime::Init() {
-  auto reader = boot_fdt::Get();
-  auto offset = reader.GetNodeOffset("/runtime");
-  auto ip = reader.GetProperty32(offset, "address");
-  auto port = reader.GetProperty16(offset, "port");
-  auto allocation_id = reader.GetProperty16(offset, "allocation_id");
+  auto cmdline = std::string(ebbrt::multiboot::CmdLine());
+  auto host = std::string("host_addr=");
+  auto loc = cmdline.find(host);
+  if (loc == std::string::npos) {
+    kabort("No host address found in command line\n");
+  }
+  host = cmdline.substr((loc + host.size()));
+  auto gap = host.find(";");
+  if (gap != std::string::npos) {
+    host = host.substr(0, gap);
+  }
+
+  auto port = std::string("host_port=");
+  loc = cmdline.find(port);
+  if (loc == std::string::npos) {
+    kabort("No host port found in command line\n");
+  }
+  port = cmdline.substr((loc + port.size()));
+  gap = port.find(";");
+  if (gap != std::string::npos) {
+    port = port.substr(0, gap);
+  }
+
+  auto idstr = std::string("allocid=");
+  loc = cmdline.find(idstr);
+  if (loc == std::string::npos) {
+    kabort("No allocation id found in command line\n");
+  }
+  idstr = cmdline.substr((loc + idstr.size()));
+  gap = idstr.find(";");
+  if (gap != std::string::npos) {
+    idstr = idstr.substr(0, gap);
+  }
+
+  auto allocation_id = atoi(idstr.c_str());
+  auto nport = atoi(port.c_str());
```

Comment:
It's not part of the C standard library. It's C++ (<string> in particular). I believe our version of GCC should support it.